backend/chunker/chunker.go

```diff
@@ -0,0 +1,1412 @@
+- Size         - total size of all chunks
+- PublicKey    - always "null", ignored
+- CreationDate - last modification time`,
+			}},
+		}, {
+			Name:     "fail_on_bad_chunks",
+			Advanced: true,
+			Default:  false,
+			Help: `The list command might encounter files with missinng or invalid chunks.
+This boolean flag tells what rclone should do in such cases.`,
+			Examples: []fs.OptionExample{
+				{
+					Value: "true",
+					Help:  "Fail with error.",
+				}, {
+					Value: "false",
+					Help:  "Silently ignore invalid object.",
+				},
+			},
+		}},
+	})
+}
+
+// NewFs constructs an Fs from the path, container:path
+func NewFs(name, rpath string, m configmap.Mapper) (fs.Fs, error) {
+	// Parse config into Options struct
+	opt := new(Options)
+	err := configstruct.Set(m, opt)
+	if err != nil {
+		return nil, err
+	}
+
+	remote := opt.Remote
+	if strings.HasPrefix(remote, name+":") {
+		return nil, errors.New("can't point remote at itself - check the value of the remote setting")
+	}
+
+	wInfo, wName, wPath, wConfig, err := fs.ConfigFs(remote)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to parse remote %q to wrap", remote)
+	}
+	// Look for a file first
+	remotePath := fspath.JoinRootPath(wPath, rpath)
+	wrappedFs, err := wInfo.NewFs(wName, remotePath, wConfig)
+	if err != fs.ErrorIsFile && err != nil {
+		return nil, errors.Wrapf(err, "failed to make remote %s:%q to wrap", wName, remotePath)
+	}
+
+	f := &Fs{
+		Fs:   wrappedFs,
+		name: name,
+		root: rpath,
+		opt:  *opt,
+	}
+
+	switch opt.MetaFormat {
+	case "none":
+		f.useMeta = false
+	case "simplejson":
+		f.useMeta = true
+	default:
+		return nil, fmt.Errorf("unsupported meta format '%s'", opt.MetaFormat)
+	}
+
+	if err := f.parseNameFormat(opt.NameFormat); err != nil {
+		return nil, fmt.Errorf("invalid name format '%s': %v", opt.NameFormat, err)
+	}
+
+	// Note 1: the features here are ones we could support, and they are
+	// ANDed with the ones from wrappedFs.
+	// Note 2: features.Fill() points features.PutStream to our PutStream,
+	// but features.Mask() will nullify it if wrappedFs does not have it.
+	f.features = (&fs.Features{
+		CaseInsensitive:         true,
+		DuplicateFiles:          true,
+		ReadMimeType:            true,
+		WriteMimeType:           true,
+		BucketBased:             true,
+		CanHaveEmptyDirectories: true,
+		SetTier:                 true,
+		GetTier:                 true,
+	}).Fill(f).Mask(wrappedFs).WrapsFs(f, wrappedFs)
+
+	return f, err
+}
+
+// Options defines the configuration for this backend
+type Options struct {
+	Remote          string        `config:"remote"`
+	ChunkSize       fs.SizeSuffix `config:"chunk_size"`
+	NameFormat      string        `config:"name_format"`
+	StartFrom       int           `config:"start_from"`
+	MetaFormat      string        `config:"meta_format"`
+	FailOnBadChunks bool          `config:"fail_on_bad_chunks"`
+}
+
+// Fs represents a wrapped fs.Fs
+type Fs struct {
+	fs.Fs
```

Comment:
You've embedded `fs.Fs` here which means you'll get its underlying methods.

This can be good or bad, however it looks like you've implemented methods you didn't need to (like `About`).

So either remove the extra methods, or make `fs.Fs` into a field.  I'd probably do the latter as then when new methods are made on the underlying backends we need to make a deliberate choice to support them here.

Reply:
For the sake of transparency I am inclined to support at least About and Cleanup. Users should not revert to underlying remote to check free space if the mostly use overlay. Yet I am less inclined to transparently wrap SetTier and whatever pops up in future.
So, I'll factor fs.Fs into a field and keep only a subset of extra methods, to my liking. The holy GoF blessed us with composition over inheritance.

Reply:
:-) composition for the win!

Reply:
Composition looses to fstests.go, as I can't drop anything:
    --- FAIL: TestIntegration/FsCheckWrap (0.00s)
        fstests.go:401: Missing Fs wrapper for DirCacheFlush
        fstests.go:401: Missing Fs wrapper for PublicLink
        fstests.go:401: Missing Fs wrapper for MergeDirs
        fstests.go:401: Missing Fs wrapper for UserInfo
        fstests.go:401: Missing Fs wrapper for Disconnect

Reply:
I now embed wrapped Fs rather than inherit (will push commits later). But I cannot remove irrelevant methods from Chunker such as DirCacheFlush, PublicLink, MergeDirs, UserInfo, Disconnect because they are required by TestIntegration/FsCheckWrap.

Reply:
You can mark these as unsupported in the test structure. See for example the test runner in the cache backend